lib/bookingsync/engine/credentials_resolver.rb

```diff
@@ -0,0 +1,18 @@
+class BookingSync::Engine::CredentialsResolver
+  attr_accessor :host
+
+  def initialize(host)
+    @host = host
+  end
+
+  def call
+    if application = ::Application.find_by_host(host)
+      {
+        client_id: application.client_id,
+        client_secret: application.client_secret
+      }
+    else
+      nil
```

Comment:
why nil? It should also be returning a hash, otherwise we have a method that returns different types for no good reason. And returning hash itself is also not ideal, best to returns some dedicated object, like `ApplicationCredentials` with a nice methods like client_id, client_secret, and possibly some extra method checking if the credentials are filled or not.